polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/EntityIdPaging.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.persistence.pagination;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import jakarta.annotation.Nullable;
+import org.apache.polaris.core.entity.PolarisBaseEntity;
+
+/** Utility class for processing pagination of streams of entities ordered by some ID property. */
+public final class EntityIdPaging {
```

Comment:
I'm not sure how useful this is going to be, since its methods are `static` there can't be another implementation. The methods both look very thin.

Reply:
`EntityIdPaging` is a specific case of constructing and parsing `PageToken.encodedDataReference()` for Persistence implementations that rely of sorted streams of entities.

Having these `static` methods allows having the encoding/decoding code across several use cases in the current Polaris code.

Reply:
I see, but I'm trying to understand what this will look like if we have `SomeOtherEntityIdPaging`. Will callers have some way of selecting a `...Paging` based on the token structure or method?

Reply:
If `SomeOtherEntityIdPaging` produces a single primitive value from a "data pointer" it would be very similar to `EntityIdPaging`, I guess.

If it produces a rich value, then it will need a companion type to hold the parsed data... but that is something to be done if and when we need it. It could be something like `static NameAndIdPaging.entityBoundary(PageToken pageToken)` returning a new `NameAndIdPaging.Boundary(name, id)`.

Reply:
> but that is something to be done if and when we need it

We have a (smaller) existing PR in #1555 that supports this use case, so I have some reservations about an alternative design that loses this functionality